libgoal/participation.go

```diff
@@ -165,8 +165,11 @@ func (c *Client) GenParticipationKeysTo(address string, firstValid, lastValid, k
 
 	// Fill the database with new participation keys
 	newPart, err := account.FillDBWithParticipationKeys(partdb, parsedAddr, firstRound, lastRound, keyDilution)
-	part = newPart.Participation
```

Comment:
This change is not necessary. 
There is something wrong here, but this change does not fix it. 
partdb should be closed. returning if err != nil leaves it open. 
There will be a fix in master, and when merged with it, should resolve it. Not making any changes is is advised.

Reply:
resolved.